src/libs/actions/IOU.ts

```diff
@@ -10981,16 +11051,19 @@ function putOnHold(transactionID: string, comment: string, reportID: string, sea
         });
     }
 
-    API.write(
-        'HoldRequest',
-        {
-            transactionID,
-            comment,
-            reportActionID: createdReportAction.reportActionID,
-            commentReportActionID: createdReportActionComment.reportActionID,
-        },
-        {optimisticData, successData, failureData},
-    );
+    const params: HoldMoneyRequestParams = {
+        transactionID,
+        comment,
+        reportActionID: createdReportAction.reportActionID,
+        commentReportActionID: createdReportActionComment.reportActionID,
+    };
+
+    if (!initialReportID) {
+        params.transactionThreadReportID = reportID;
+        params.createdReportActionIDForThread = optimisticCreatedAction.reportActionID;
+    }
+
+    API.write('HoldRequest', params, {optimisticData, successData, failureData});
```

Comment:
@VickyStash From the diff, I can see you are not the original one to blame for this, but since you are the last one that touched this, can you please make a quick PR to ensure this is referencing the constant `HOLD_MONEY_REQUEST`?

Reply:
Looks like the same thing happened to `'UnHoldRequest'` as well

Reply:
Yes. It came from https://github.com/Expensify/App/pull/33897

Reply:
Anyway, I created a PR here:
https://github.com/Expensify/App/pull/67474

cc @iwiznia

Reply:
Oh sorry for this, should've caught it.